pkg/connector/kairosdb.go

```diff
@@ -0,0 +1,568 @@
+		return fmt.Errorf("kairosdb[%s]: invalid HTTP backend response: %s", connector.name, err)
+	}
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to read HTTP response body: %s", connector.name, err)
+	}
+	if err = json.Unmarshal(data, &JSONmetrics); err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to unmarshal JSON data: %s", connector.name, err)
+	}
+
+        metrics := make([]map[string]string, 0)
+	for _, m := range JSONmetrics["results"] {
+		metrics = append(metrics, map[string]string{"name": m})
+	}
+        query := map[string]interface{}{"metrics": metrics}
+	if connector.startAbsolute > 0 {
+		query["start_absolute"] = connector.startAbsolute
+	} else {
+		query["start_relative"] = connector.startRelative
+	}
+	if connector.endAbsolute > 0 {
+		query["end_absolute"] = connector.endAbsolute
+	}
+	if connector.endRelative != nil {
+		query["end_relative"] = connector.endRelative
+	}
+	jsonData, err := json.Marshal(query)
+	if err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to marshal JSON data: %s", connector.name, err)
+	}
+
+	logger.Log(logger.LevelDebug, "connector", "kairosdb[%s]: API Call to %s: %s", connector.name,
+		strings.TrimSuffix(connector.URL, "/")+kairosdbURLMetricsTags,
+		string(jsonData))
+
+	request, err = http.NewRequest("POST", strings.TrimSuffix(connector.URL, "/")+kairosdbURLMetricsTags, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to set up HTTP request: %s", connector.name, err)
+	}
+
+	request.Header.Add("User-Agent", "Facette")
+	request.Header.Add("X-Requested-With", "KairosDBConnector")
+	request.Header.Set("Content-Type", "application/json")
+
+	response, err = httpClient.Do(request)
+	if err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to perform HTTP request: %s", connector.name, err)
+	}
+	defer response.Body.Close()
+
+	if err = kairosdbCheckBackendResponse(response); err != nil {
+		return fmt.Errorf("kairosdb[%s]: invalid HTTP backend response: %s", connector.name, err)
+	}
+
+	data, err = ioutil.ReadAll(response.Body)
+	if err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to read HTTP response body: %s", connector.name, err)
+	}
+
+	logger.Log(logger.LevelDebug, "connector", "kairosdb[%s]: API Response from %s: %s", connector.name,
+		strings.TrimSuffix(connector.URL, "/")+kairosdbURLMetricsTags,
+		string(data))
+
+	if err = json.Unmarshal(data, &JSONquery); err != nil {
+		return fmt.Errorf("kairosdb[%s]: unable to unmarshal JSON data: %s", connector.name, err)
+	}
+
+	for _, q := range JSONquery["queries"] {
+		for _, r := range q["results"] {
+			metricName := r.Name
+			aggregator := matchAggregatorPattern(connector.aggregators, metricName)
+			if aggregator == nil {
+				aggregator = connector.defAggregator
+			}
+
+			for _, t := range connector.srcTags {
+				if _, ok := r.Tags[t]; !ok {
+					continue
+				}
+				var sc uint64
```

Comment:
Any actual use to this counter, besides to be displayed in the log message below?